pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -192,6 +200,36 @@ func (c *Controller) createNewBackup(cluster *v3.Cluster) (*v3.EtcdBackup, error
 
 }
 
+func (c *Controller) etcdSaveWithBackoff(b *v3.EtcdBackup) (runtime.Object, error) {
+	backoff := wait.Backoff{
+		Duration: 1000 * time.Millisecond,
+		Factor:   2,
+		Jitter:   0,
+		Steps:    5,
+	}
+	kontainerDriver, err := c.KontainerDriverLister.Get("", service.RancherKubernetesEngineDriverName)
+	if err != nil {
+		return b, err
+	}
+
+	return v3.BackupConditionCompleted.Do(b, func() (runtime.Object, error) {
+		cluster, err := c.clusterClient.Get(b.Spec.ClusterID, metav1.GetOptions{})
+		if err != nil {
+			return b, err
+		}
+		var inErr error
+		err = wait.ExponentialBackoff(backoff, func() (bool, error) {
+			if inErr = c.backupDriver.ETCDSave(context.Background(), cluster.Name, kontainerDriver, cluster.Spec, b.Name); err != inErr {
```

Comment:
I want to understand why are we retrying the whole ETCDSave function? Are there some errors that this function can return for which we can just return?

Reply:
Let's use context of the controller instead of background context